apps/emqx_bridge/src/emqx_bridge_resource.erl

```diff
@@ -340,6 +340,8 @@ parse_confs(Type, Name, Conf) when ?IS_INGRESS_BRIDGE(Type) ->
 %% to hocon; keeping this as just `kafka' for backwards compatibility.
 parse_confs(<<"kafka">> = _Type, Name, Conf) ->
     Conf#{bridge_name => Name};
+parse_confs(<<"pulsar_producer">> = _Type, Name, Conf) ->
```

Comment:
I know, over-engineering and not for this PR but this could be sth like:

```erlang
bridge_impl_module(BridgeType) ->
  binary_to_existing_atom(<<"emqx_bridge_", BridgeType/binary, "_impl").

parse_confs(Type, _Name, BridgeConfig) ->
  M = bridge_impl_module(Type), 
  M:connector_config(BridgeConfig).
```
I.e. `emqx_bridge` defines a behavior and this behavior needs to be implemented by a bridge_impl module. But could also be an optional callback, then more like

```erlang
parse_confs(Type, _Name, BridgeConfig) ->
  M = bridge_impl_module(Type), 
  case erlang:function_exported(M, connector_config, 1) of
    true ->
       M:connector_config(BridgeConfig);
    false ->
       BridgeConfig
  end.
```

Reply:
Similar for "is_ingress_bridge" and "is_egress_bridge" in `emqx_bridge` and `emqx_bridge_resource` - those should be callbacks into the implementation module rather than hardcoding it into the generic parts.

Reply:
That sound like a good idea.  Indeed, I think making that change would fit better in a separate PR.